executor/cleanup/cleanup.go

```diff
@@ -86,81 +83,8 @@ func cleanup(client *kubernetes.Clientset, namespace string, instanceId string)
 	return nil
 }
 
-// TODO: Since different executor type has different idleObjectReaper strategy, move this part to executor type for better code separation.
-// idleObjectReaper reaps objects after certain idle time
-func idleObjectReaper(kubeClient *kubernetes.Clientset,
-	fissionClient *crd.FissionClient,
-	fsCache *fscache.FunctionServiceCache,
-	idlePodReapTime time.Duration) {
-
-	pollSleep := time.Duration(2 * time.Minute)
-	for {
-		time.Sleep(pollSleep)
-
-		envs, err := fissionClient.Environments(meta_v1.NamespaceAll).List(meta_v1.ListOptions{})
-		if err != nil {
-			log.Fatalf("Failed to get environment list: %v", err)
-		}
-
-		envList := make(map[types.UID]struct{})
-		for _, env := range envs.Items {
-			envList[env.Metadata.UID] = struct{}{}
-		}
-
-		funcSvcs, err := fsCache.ListOld(idlePodReapTime)
-		if err != nil {
-			log.Printf("Error reaping idle pods: %v", err)
-			continue
-		}
-
-		for _, fsvc := range funcSvcs {
-			if _, ok := envList[fsvc.Environment.Metadata.UID]; !ok {
-				log.Printf("Environment %v for function %v no longer exists",
-					fsvc.Environment.Metadata.Name, fsvc.Name)
-			}
-
-			if fsvc.Environment.Spec.AllowedFunctionsPerContainer == fission.AllowedFunctionsPerContainerInfinite {
-				continue
-			}
-
-			fn, err := fissionClient.Functions(fsvc.Function.Namespace).Get(fsvc.Function.Name)
-			if err != nil {
-				// Newdeploy manager handles the function delete event and clean cache/kubeobjs itself,
-				// so we ignore the not found error for functions with newdeploy executor type here.
-				if errors.IsNotFound(err) && fsvc.Executor == fscache.NEWDEPLOY {
-					continue
-				}
-				log.Printf("Error getting function: %v", fsvc.Function.Name)
-				continue
-			}
-
-			// Ignore functions of NewDeploy ExecutorType with MinScale > 0
-			if fn.Spec.InvokeStrategy.ExecutionStrategy.MinScale > 0 &&
-				fn.Spec.InvokeStrategy.ExecutionStrategy.ExecutorType == fission.ExecutorTypeNewdeploy {
-				continue
-			}
-
-			deleted, err := fsCache.DeleteOld(fsvc, idlePodReapTime)
-			if err != nil {
-				log.Printf("Error deleting Kubernetes objects for fsvc '%v': %v", fsvc, err)
-				log.Printf("Object Name| Object Kind | Object Namespace")
-				for _, kubeobj := range fsvc.KubernetesObjects {
-					log.Printf("%v | %v | %v", kubeobj.Name, kubeobj.Kind, kubeobj.Namespace)
-				}
-			}
-
-			if !deleted {
-				continue
-			}
-
-			for _, kubeobj := range fsvc.KubernetesObjects {
-				deleteKubeobject(kubeClient, &kubeobj)
-			}
-		}
-	}
-}
-
-func deleteKubeobject(kubeClient *kubernetes.Clientset, kubeobj *apiv1.ObjectReference) {
+// DeleteKubeObject deletes given kubernetes object
+func DeleteKubeObject(kubeClient *kubernetes.Clientset, kubeobj *apiv1.ObjectReference) {
```

Comment:
Just wondering, why is this called `DeleteX` when all other functions in this pkg are `CleanupX`?

Reply:
I didn't notice it before. Rename to `CleanupKubeObject`